### ISSUE TITLE
Question: retrain.py: Addition to read in file list (with data set split) and confusion matrix output

### ISSUE THREAD

**Issue by andreas-seiderer** [tensorflow/hub#182]
Hello,

We conducted some image transfer learning experiments with four embedded / mobile ARM devices (CPU only) and several models where we used an adapted version of the retrain.py script.

We wrote a paper which is not available yet but we have a companion repository with the plots / measurements / used scripts to allow reproduction / simplify further tests: https://github.com/hcmlab/tf-transferlearning-benchmarks

Some of the changes done to the retrain.py script were just to add additional logging but we also required a function allowing us to read in a file list with adapted training / validation and test splits ([retrain_hub_filelists.py](https://github.com/hcmlab/tf-transferlearning-benchmarks/blob/master/scripts/retrain_hub_filelists.py#L155)). Example file: [flower_photos_list_default.tsv](https://github.com/hcmlab/tf-transferlearning-benchmarks/blob/master/scripts/lists/flower_photos_list_default.tsv)
I added an additional parameter which can be used instead of "image_dir".

Additionally, I added a file output of a confusion matrix to allow a more detailed evaluation (e.g. recall / precision) of the test set: [code](https://github.com/hcmlab/tf-transferlearning-benchmarks/blob/master/scripts/retrain_hub_filelists.py#L921), [example file](https://github.com/hcmlab/tf-transferlearning-benchmarks/blob/master/measurements/tf1.8_incremental_tests/all_increment/1/mobilenet_v2_100_128_all_increment_100_1000.txt_eval.txt)

Although the script is intended as an example, I think these additions could also be of use for others. Is there interest on your side to add these features to the current version of the official retrain.py script?

**Comment by andreas-seiderer** [comment]
Hi, I mainly used this example as it is currently one of the most sophisticated transfer learning examples I could find, where you can choose easily from a wide range of models and have sufficient control over the retraining process without having to code everything from the ground up (I also looked at other frameworks). For the feasibility tests I conducted it did a good job with the adaptions :) . 
Mainly, I wanted to let you know that there is an extended version available but I can fully understand that you prefer not to increase the code length as this makes it even more complex.
